test/integration/subprocess_mixin.py

```diff
@@ -0,0 +1,74 @@
+"""Mixin for managing child subprocess with an additional thread."""
+
+from threading import Thread, Condition
+import subprocess
+import logging
+from abc import ABC, abstractmethod
+
+
+class SubprocessMixin(ABC):
+  """Mixin used to manage launching subprocess using a separate Python thread."""
+
+  def __init__(self):
+    """Create SubprocessMixin."""
+    self._server_thread = Thread(target=self._serverThreadRunner)
+    self._server_process = None
+    self._has_launched = False
+    self._has_launched_cv = Condition()
+    self.stdout = b''
```

Comment:
Do we have to store the stderr and stdout as class attributes? The lifetime feels questionable, since the class could potentially be reused for multiple runs.

The `_serverThreadRunner` returns the same, can we just return them from a public method without storing them?

Reply:
makes sense, done.